src/core/src/enclave_sgx.rs

```diff
@@ -447,3 +447,90 @@ fn check_verification_report(
+
+        let ret = enclave.get_epid_group();
+        assert_eq!((ret.len() > 0), true);
+    }
+
+    #[test]
+    fn test_is_sgx_simulator() {
+        let mut enclave = EnclaveConfig::default();
+        enclave.initialize_enclave();
+        assert_eq!( (enclave.enclave_id.handle > 0), true);
+
+        let ret = enclave.check_if_sgx_simulator();
+        println!("is_sgx_simulator ? {:?}", ret);
+    }
+
+    #[test]
+    fn test_create_signup_info() {
+        let mut enclave = EnclaveConfig::default();
+        enclave.initialize_enclave();
+        assert_eq!( (enclave.enclave_id.handle > 0), true);
+
+        let pub_key_hash = vec![0x45; 32];
+        enclave.create_signup_info(&pub_key_hash);
+        assert_eq!( (enclave.signup_info.handle > 0), true);
+    }
+
+    #[test]
+    fn test_create_wait_certificate() {
+        let mut enclave = EnclaveConfig::default();
+        enclave.initialize_enclave();
+        assert_eq!( (enclave.enclave_id.handle > 0), true);
+
+        let pub_key_hash = vec![0x45; 32];
+        enclave.create_signup_info(&pub_key_hash.to_vec());
+        assert_eq!( (enclave.signup_info.handle > 0), true);
+
+        let mut duration: u64 = 0x0102030405060708;
+        let prev_cert = "".to_string();
+        let prev_wait_cert_sig = "".to_string();
+        let validator_id = vec![0x41; 32];
+        let prev_block_id = "abc".to_string();
+        let block_summary = "this is first block".to_string();
+        let wait_time = 10_u64;
+
+        let (poet_pub_key, enclave_quote) = enclave.get_signup_parameters();
+
+        let ret_dur = EnclaveConfig::initialize_wait_certificate(
+                                            enclave.enclave_id,
+                                            prev_cert.clone(),
+                                            prev_wait_cert_sig.clone(),
+                                            &validator_id, &poet_pub_key);
+        assert_eq!((ret_dur > 0), true);
+
+        let (wait_cert, wait_cert_sig) = EnclaveConfig::finalize_wait_certificate(
+                                                    enclave.enclave_id,
+                                                    prev_cert.clone(),
+                                                    prev_block_id.clone(),
+                                                    prev_wait_cert_sig.clone(),
+                                                    block_summary.clone(),
+                                                    wait_time.clone());
+        assert_eq!(wait_cert.is_empty(), false);
+        assert_eq!(wait_cert_sig.is_empty(), false);
+        println!("wait cert = {}", wait_cert);
+        let verify_status = EnclaveConfig::verify_wait_certificate(
+                                        enclave.enclave_id, &poet_pub_key,
+                                        &wait_cert, &wait_cert_sig);
+        println!("Is verification of wait certificate Passed: {}", verify_status);
```

Comment:
check verify_status and remove println

Reply:
i cannot use assert to check status as it would mark the test case as failed if the verification of wait certificate fails (without any issues).